src/Stratis.Bitcoin.Features.Consensus/Rules/CommonRules/HeaderTimeChecksRule.cs

```diff
@@ -0,0 +1,42 @@
+﻿using System;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using NBitcoin;
+using Stratis.Bitcoin.Consensus.Rules;
+using Stratis.Bitcoin.Utilities;
+
+namespace Stratis.Bitcoin.Features.Consensus.Rules.CommonRules
+{
+    /// <summary>
+    /// Check time .
+    /// </summary>
+    [HeaderValidationRule(CanSkipValidation = true)]
+    public class HeaderTimeChecksRule : ConsensusRule
+    {
+        /// <inheritdoc />
+        /// <exception cref="ConsensusErrors.TimeTooOld">Thrown if block's timestamp is too early.</exception>
+        /// <exception cref="ConsensusErrors.TimeTooNew">Thrown if block' timestamp too far in the future.</exception>
+        public override Task RunAsync(RuleContext context)
+        {
+            Guard.NotNull(context.ConsensusTip, nameof(context.ConsensusTip));
+
+            BlockHeader header = context.ValidationContext.Block.Header;
+
+            // Check timestamp against prev.
+            if (header.BlockTime <= context.ConsensusTip.GetMedianTimePast())
+            {
+                this.Logger.LogTrace("(-)[TIME_TOO_OLD]");
+                ConsensusErrors.TimeTooOld.Throw();
+            }
+
+            // Check timestamp.
+            if (header.BlockTime > (context.Time + TimeSpan.FromHours(2)))
```

Comment:
This is wrong, there must be a variable that is initialised for the specific network

Reply:
This is supposed to be future drift– See the other rule you have below

Reply:
future drift is a pos rule its in the other rule `StratisBigFixPosFutureDriftRule`

Reply:
You are saying this is not a POS rule?
I can rename it to PowFutureFrift I guess.

Reply:
Correct